### PR TITLE
CSL-7: Add CSL app skeleton

### PR DESCRIPTION
**Pull request by dk4g**
### What?
[CSL-7](https://collaboration.homeoffice.gov.uk/jira/browse/CSL-7)

Adds the necessary files to create the Controlled Substance Licence (CSL) - Precursor Chemical Application form skeleton. This includes the Node.js/HOF and Docker config files needed to build and run a HOF app, plus the basic folder structure for the CSL app itself with minimal content added.

Also includes housekeeping files such as README, pull request template and ignore files for Docker and Git.

### Why?
This skeleton should allow a developer to run a local version of this app with either Node.js or docker-compose, and work on page specific tickets for the form.

### Anything else?
This PR does not include HOF devops setup such as the /bin, /kube folders or drone.yml. These can be appended to this feature or added as a new feature later. The purpose of this ticket is to enable local development of form pages.

### CODE REVIEW

File: config.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    hostname: process.env.FILE_VAULT_URL,
+    allowedMimeTypes: [
+      'image/jpeg',
+      'image/jpg',
+      'image/png',
+      'application/pdf',
+      'application/msword',
+      'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
+      'application/vnd.ms-excel',
+      'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet'
```
Comment (jamiecarterHO) [inline config.js:29]: Do we have any confirmation on this list yet?

File: hof.settings.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  "theme": "govUK",
+  "behaviours": [
+    "hof/components/clear-session"
+  ],
+  "translations": "./apps/pca/translations",
+  "routes": [
+    "./apps/pca"
+  ],
+  "session": {
+    "name": "pca.hof.sid"
```
Comment (jamiecarterHO) [inline hof.settings.json:12]: other multi-app forms use a generic name here `csl.hof.sid` perhaps.

**Review by jamiecarterHO — COMMENTED** [review]
Few small questions

**Comment by RobertMcCann** [comment]
we could also add an empty section folder too for summary data sections?

File: README.md [hunk 3]
```diff
@@ -1 +1,95 @@
...
+- [File Vault](https://github.com/UKHomeOffice/file-vault) to store and retrieve uploaded files
+
+## Install & Run the Application locally
+
+### Prerequisites
+
+- [Node.js](https://nodejs.org/en/) - v.20.18.0
+- [Redis server](http://redis.io/download) running on default port 6379
+- [File Vault](https://github.com/UKHomeOffice/file-vault) Service - running port 3000
+
```
[the remark this answers is not in the document]
Comment (dk4g) [reply README.md:29]: Agreed, this is an initial version to make it easy for anyone to run on the local environment and at this stage we will continue to update it over time

File: config.js (hunk 1 above)
Comment (dk4g) [reply config.js:29]: `a PNG, PDF, JPEG or Microsoft Word or Excel file`
the requirements are a bit loose, I included `JPG` as well and raised it with BA to confirm,
For Microsoft Word Excel the extensions included are - doc, docx, xls, xlsx

**Comment by dk4g** [comment]
> we could also add section folder and summary-data-sections.js for summary data sections?
This can be covered by the separate PR in more details in order to keep this one as small as possible

File: config.js (hunk 1 above)
Comment (jamiecarterHO) [reply config.js:29]: Sure, we can refine from here when we have confirmation

**Review by jamiecarterHO — APPROVED** [review]
Looks good as a start :)

**Review by vinodhasamiyappanHO — APPROVED** [review]
Looks good to me

**Pull request merged**